From: NeilBrown <neilb@suse.com>
Subject: kabi fix for sunrpc-use-after-free-in-svc_process_common.patch
Patch-mainline: never, kabi
References: bsc#1119946 CVE-2018-16884

'struct svc_rqst' is visible to modules -  auth_gss in particular,
so changes must be hidden for kabi stability.
The new field is not useful to an auth module, so hiding it
won't hurt.

Signed-off-by: NeilBrown <neilb@suse.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 include/linux/sunrpc/svc.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/sunrpc/svc.h
+++ b/include/linux/sunrpc/svc.h
@@ -291,9 +291,11 @@ struct svc_rqst {
 	struct svc_cacherep *	rq_cacherep;	/* cache info */
 	struct task_struct	*rq_task;	/* service thread */
 	spinlock_t		rq_lock;	/* per-request lock */
+#ifndef __GENKSYMS__
 	struct net		*rq_bc_net;	/* pointer to backchannel's
 						 * net namespace
 						 */
+#endif
 };
 
 #define SVC_NET(rqst) (rqst->rq_xprt ? rqst->rq_xprt->xpt_net : rqst->rq_bc_net)
